### PR TITLE
Highlight related edges and cardinalities when a TableNode is active.

### PR DESCRIPTION
**Pull request by FunamaYukina**
## Summary

Highlight related edges and cardinalities when a TableNode is active.
Preview: https://liam-erd-sample-gx3t0p2yn-route-06-core.vercel.app/

### Before
ss 2506

### After
ss 2507

### CODE REVIEW

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/TableNode/type.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export type Data = {
   highlightedHandles: string[]
   sourceColumnName: string | undefined
   targetColumnCardinalities?: Record<string, Cardinality> | undefined
+  onClick?: (tableName: string) => void
```
Comment (MH4GF) [inline frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/TableNode/type.ts:11]: This does not appear to have been used from TableNode, was it necessary?

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx [hunk 2]
```diff
@@ -65,47 +65,66 @@ export const ERDContentInner: FC<Props> = ({
   const {
     state: { loading },
   } = useERDContentContext()
+  const [clickedNodeId, setClickedNodeId] = useState<string | null>(null)
```
Comment (MH4GF) [inline frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx:68]: It is painful to have to create a separate state from the `active` state managed by `useUserEditingStore()`, but it seems that the logic has to be unravelled to solve the problem. I'd like to try another PR.

**Review by MH4GF — APPROVED** [review]
The highlighting process is getting more complicated, but it looks good!

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx (hunk 2 above)
Comment (FunamaYukina) [reply frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx:68]: You're absolutely right. I wanted to write the code more simply, but I couldn't. In the end, the code became more complex🥲.Thank you for thinking about it🙏.

[the remark this answers is not in the document]
Comment (hoshinotsuyoshi) [reply frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx:68]: I agree that it is painful. I was thinking about checking whether the changes we made to how we handle data around PR #280 could be utilized. The functionality seems good, so I think it's fine to merge it now and integrate later!

**Review by hoshinotsuyoshi — APPROVED** [review]

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/TableNode/type.ts (hunk 1 above)
Comment (FunamaYukina) [reply frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/TableNode/type.ts]: Oh,sorry, you are right.
I've removed the unnecessary props🙏. https://github.com/liam-hq/liam/pull/284/commits/a518be68e426f6ad81da6cd613249c18448c2738

File: frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (FunamaYukina) [reply frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/ERDContent.tsx]: Thank you for your suggestion! I agree!
I've fixed it. https://github.com/liam-hq/liam/pull/284/commits/9c8767ee97a49d4aae09eb5fbc4eacfe95605b7a

**Comment by FunamaYukina** [comment]
Thank you all for your reviews.🙏

**Pull request merged**